### DISCUSSION TITLE
UX improvements

### DISCUSSION THREAD

**Discussion by gemueselasagne** [icflorescu/mantine-contextmenu#160] [Ideas]
Good read on how the Adobe team improved their context menu UX! Anyone willing to bring this to mantine-contextmenu? 🔥

👉🏻 https://react-spectrum.adobe.com/blog/creating-a-pointer-friendly-submenu-experience.html

**Comment by icflorescu** [comment]
Hey, @gemueselasagne, thanks for bringing this up.
An interesting read indeed, and I couldn't agree more. There's nothing more annoying that a submenu that disappears when you're trying to reach its content with your mouse cursor 🙂
I'll admit there's room for improvement in the [present implementation of Mantine ContextMenu](https://icflorescu.github.io/mantine-contextmenu/examples/submenus/).
I'm currently using a simple timeout/delay strategy to ensure a minimal level of UX comfort, and that proved to be enough so far, but I'm 100% open to a PR addressing this.
The relevant code is [here](https://github.com/icflorescu/mantine-contextmenu/blob/main/package/ContextMenuItem.tsx), feel free to play around with it if you feel like giving it a try 🙂
